services/fluxmonitor.go

```diff
@@ -354,9 +353,46 @@ func (fm *FluxMonitor) getAdapterResponse(endpoint url.URL, requestData string)
 	return response.Price, json.Unmarshal(body, &response)
 }
 
-func (fm *FluxMonitor) poll() error {
+// PollAndGetAnswer() will poll if the latest answer is older than the polling interval
+// and then return the latest value
+func (fm *FluxMonitor) PollAndGetAnswer() (decimal.Decimal, error) {
 	fm.pollMutex.Lock()
 	defer fm.pollMutex.Unlock()
+
+	if time.Since(fm.latestResultTimestamp)+fm.config.AdapterTimeout <= fm.config.PollInterval {
+		// The result we have is from within our polling interval, so we can use it
+		return fm.latestResult, nil
+	}
+
+	return fm.pollWithRetry()
+}
+
+func (fm *FluxMonitor) ForceNewPoll() error {
+	fm.pollMutex.Lock()
+	defer fm.pollMutex.Unlock()
+
+	_, err := fm.pollWithRetry()
+	return err
+}
+
+func (fm *FluxMonitor) pollWithRetry() (decimal.Decimal, error) {
+	attempts := 3
```

Comment:
Good idea! Maybe move the interval & attempts somewhere as config? At least as constants on top.